### DISCUSSION TITLE
Is there a rule to prefer explicit boolean checks ?

### DISCUSSION THREAD

**Discussion by AbdealiLoKo** [astral-sh/ruff#7941] [Q&A]
I have a rule in eslint which checks in Typescript when I am using truthy/falsey values - and forces me to provide a better condition.

This encourages people to be more explicit in their condition: `val = 0; if(val)` should be changed to: `val = 0; if(val !== null)`

I was wondering if there is something similar in `ruff` ?

I am one of those people who prefers `if val is None:` instead of `if not val:` :)
The closest rule I have seen when I searched around was: https://docs.astral.sh/ruff/rules/compare-to-empty-string/

**Comment by hamedrabah** [comment]
Ruff does not currently have a direct equivalent of TypeScript ESLint's `strict-boolean-expressions` rule.

Python deliberately defines truth-value testing for many types, so a blanket rule requiring `x is not None`, `len(x) > 0`, and so on would reject a large amount of idiomatic Python. Ruff's boolean-related rules cover narrower problems—for example boolean positional arguments and simplifiable comparisons—but they do not use type information to decide whether `if value:` is acceptable.

For the specific “optional value accidentally treated as false” class, a type checker is the better enforcement point. Enable strict optional checking in mypy or Pyright and use explicit `is None` / `is not None` where zero or an empty container is a valid value. If you need the ESLint policy across all types, it would require a type-aware custom lint rule; Ruff's existing syntax-only rule set cannot reliably distinguish `bool`, `Optional[int]`, and user-defined truthiness in that expression.
